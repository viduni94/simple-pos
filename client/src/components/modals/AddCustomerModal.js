import React, { Component } from "react";
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from "reactstrap";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { addCustomer } from "../../actions/customerActions";
import classnames from "classnames";

class AddCustomerModal extends Component {
  constructor(props) {
    super(props);

    this.state = {
      fname: "",
      lname: "",
      mobile: "",
      userId: this.props.auth.user.id,
      error: {}
    };

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.toggleModalClose = this.toggleModalClose.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.errors) {
      this.setState({ errors: nextProps.errors });
    }
  }

  toggleModalClose = () => {
    this.setState({ fname: "", lname: "", mobile: "" });
  };

  onSubmit(e) {
    e.preventDefault();

    const customerData = {
      fname: this.state.fname,
      lname: this.state.lname,
      mobile: this.state.mobile,
      userId: this.state.userId
    };
    this.props.addCustomer(customerData);
    this.setState({ fname: "", lname: "", mobile: "" });
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  render() {
    const { user } = this.props.auth;

    return (
      <div>
        <Modal isOpen={this.props.toggle.modal} toggle={this.props.toggle.toggle} backdrop={this.props.toggle.backdrop}>
          <ModalHeader toggle={this.props.toggle.toggle}>Add New Customer</ModalHeader>
          <form onSubmit={this.onSubmit}>
            <ModalBody>
              <div className="form-group">
                <input type="text" className={classnames("form-control form-control-lg")} name="fname" value={this.state.fname} onChange={this.onChange} required />
                <small className="form-text text-muted">First Name *</small>
                {/* {errors.fname && <div className="invalid-feedback">{errors.fname}</div>} */}
              </div>
              <div className="form-group">
                <input type="text" className="form-control form-control-lg" name="lname" value={this.state.lname} onChange={this.onChange} />
                <small className="form-text text-muted">Last Name</small>
              </div>
              <div className="form-group">
                <input type="text" className={classnames("form-control form-control-lg")} name="mobile" value={this.state.mobile} onChange={this.onChange} required />
                <small className="form-text text-muted">Mobile Number *</small>
                {/* {errors.mobile && <div className="invalid-feedback">{errors.mobile}</div>} */}
              </div>
              <div className="form-group">
                <input type="hidden" className={classnames("form-control form-control-lg")} name="userId" value={user.id} required />
              </div>
            </ModalBody>
            <ModalFooter>
              <Button type="reset" onClose={this.toggleModalClose} color="secondary" onClick={this.props.toggle.toggle}>
                Cancel
              </Button>
              <Button type="submit" color="success" onClick={this.props.toggle.toggle}>
                Save
              </Button>
            </ModalFooter>
          </form>
        </Modal>
      </div>
    );
  }
}

AddCustomerModal.propTpes = {
  addCustomer: PropTypes.func.isRequired,
  customer: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  auth: state.auth,
  customer: state.customer,
  errors: state.errors
});

export default connect(
  mapStateToProps,
  { addCustomer }
)(AddCustomerModal);
