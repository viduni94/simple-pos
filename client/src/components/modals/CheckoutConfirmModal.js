import React, { Component } from "react";
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from "reactstrap";
import { connect } from "react-redux";
import { checkoutOrder } from "../../actions/orderListActions";
import PropTypes from "prop-types";
import { withRouter } from "react-router-dom";

export class CheckoutConfirmModal extends Component {
  constructor(props) {
    super(props);

    this.onSubmit = this.onSubmit.bind(this);
  }

  onSubmit(e) {
    e.preventDefault();

    const orderId = this.props.orderList.activeOrder;
    this.props.checkoutOrder(orderId);
  }

  render() {
    return (
      <div>
        <Modal isOpen={this.props.toggle.modal} toggle={this.props.toggle.toggle} backdrop={this.props.toggle.backdrop}>
          <ModalHeader toggle={this.props.toggle.toggle}>Checkout Confirmation</ModalHeader>
          <form onSubmit={this.onSubmit}>
            <ModalBody>
              <p>Are you sure you want to checkout this order?</p>
            </ModalBody>
            <ModalFooter>
              <Button type="reset" onClose={this.toggleModalClose} color="secondary" onClick={this.props.toggle.toggle}>
                No
              </Button>
              <Button type="submit" color="danger" onClick={this.props.toggle.toggle}>
                Yes
              </Button>
            </ModalFooter>
          </form>
        </Modal>
      </div>
    );
  }
}

CheckoutConfirmModal.propTypes = {
  orderList: PropTypes.object.isRequired,
  checkoutOrder: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  orderList: state.orderList
});

export default connect(
  mapStateToProps,
  { checkoutOrder }
)(withRouter(CheckoutConfirmModal));
