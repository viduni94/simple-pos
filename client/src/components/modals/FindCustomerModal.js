import React from "react";
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from "reactstrap";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import Select from "react-select";
import { getAllCustomers, setActiveCustomer } from "../../actions/customerActions";
import { withRouter } from "react-router-dom";

class FindCustomerModal extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      mobile: "",
      errors: {},
      customerId: ""
    };

    this.onSubmit = this.onSubmit.bind(this);
    this.toggleModalClose = this.toggleModalClose.bind(this);
  }

  toggleModalClose = () => {
    this.setState({ mobile: "" });
  };

  onSubmit(e) {
    e.preventDefault();

    const activeCustomer = this.props.customer.customers.filter(cus => cus._id === this.state.customerId);
    localStorage.setItem("activeCustomer", JSON.stringify(activeCustomer[0]));
    this.props.setActiveCustomer(activeCustomer[0]);
  }

  componentDidMount() {
    this.props.getAllCustomers();
  }

  render() {
    const { customers } = this.props.customer;

    const mobileNumbers = []; 

    customers.map(customer => {
      const obj = {
        label: customer.mobile,
        value: customer._id
      };
      return mobileNumbers.push(obj);
    });

    return (
      <div>
        <Modal isOpen={this.props.toggle.modal} toggle={this.props.toggle.toggle} backdrop={this.props.toggle.backdrop}>
          <ModalHeader toggle={this.props.toggle.toggle}>Find Customer</ModalHeader>
          <form onSubmit={this.onSubmit}>
            <ModalBody>
              <div className="form-group">
                <Select name="mobile" options={mobileNumbers} onChange={opt => this.setState({ customerId: opt.value })} required />
                <small className="form-text text-muted">Select customer from mobile</small>
              </div>
            </ModalBody>
            <ModalFooter>
              <Button type="reset" onClose={this.toggleModalClose} color="secondary" onClick={this.props.toggle.toggle}>
                Cancel
              </Button>
              <Button type="submit" color="success" onClick={this.props.toggle.toggle}>
                Select Customer
              </Button>
            </ModalFooter>
          </form>
        </Modal>
      </div>
    );
  }
}

FindCustomerModal.propTypes = {
  customer: PropTypes.object.isRequired,
  getAllCustomers: PropTypes.func.isRequired,
  setActiveCustomer: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  customer: state.customer
});

export default connect(
  mapStateToProps,
  { getAllCustomers, setActiveCustomer }
)(withRouter(FindCustomerModal));
