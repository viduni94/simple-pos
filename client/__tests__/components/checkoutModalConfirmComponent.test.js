import React from "react";
import { shallow } from "enzyme";
import { CheckoutConfirmModal } from "../../src/components/modals/CheckoutConfirmModal";

const props = {
  orderList: {},
  checkoutOrder: jest.fn().mockReturnThis(),
  toggle: {modal: "", toggle: "", backdrop: true}
};

describe("Checout modal Component", () => {
    test("renders", done => {
      const wrapper = shallow(<CheckoutConfirmModal {...props} />);
      expect(wrapper).toHaveLength(1);
      done();
    });
  
    test("should render initial layout", () => {
      // when
      const component = shallow(<CheckoutConfirmModal {...props} />);
      // then
      expect(component.getElements()).toMatchSnapshot();
    });

})