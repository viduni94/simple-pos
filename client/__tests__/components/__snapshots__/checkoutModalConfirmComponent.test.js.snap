// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checout modal Component should render initial layout 1`] = `
Array [
  <div>
    <Modal
      autoFocus={true}
      backdrop={true}
      backdropTransition={
        Object {
          "mountOnEnter": true,
          "timeout": 150,
        }
      }
      centered={false}
      fade={true}
      isOpen=""
      keyboard={true}
      modalTransition={
        Object {
          "timeout": 300,
        }
      }
      onClosed={[Function]}
      onOpened={[Function]}
      role="dialog"
      toggle=""
      zIndex={1050}
    >
      <ModalHeader
        charCode={215}
        closeAriaLabel="Close"
        tag="h5"
        toggle=""
        wrapTag="div"
      >
        Checkout Confirmation
      </ModalHeader>
      <form
        onSubmit={[Function]}
      >
        <ModalBody
          tag="div"
        >
          <p>
            Are you sure you want to checkout this order?
          </p>
        </ModalBody>
        <ModalFooter
          tag="div"
        >
          <Button
            color="secondary"
            onClick=""
            tag="button"
            type="reset"
          >
            No
          </Button>
          <Button
            color="danger"
            onClick=""
            tag="button"
            type="submit"
          >
            Yes
          </Button>
        </ModalFooter>
      </form>
    </Modal>
  </div>,
]
`;
